packages/legend-extension-dsl-data-space/src/components/query/DataSpaceQueryBuilder.tsx

```diff
@@ -408,3 +419,46 @@ export const renderDataSpaceQueryBuilderSetupPanelContent = (
     queryBuilderState={queryBuilderState}
   />
 );
+
+export const queryDataSpace = async (
+  dataSpace: DataSpace,
+  editorStore: EditorStore,
+): Promise<void> => {
+  const embeddedQueryBuilderState = editorStore.embeddedQueryBuilderState;
+  await flowResult(
+    embeddedQueryBuilderState.setEmbeddedQueryBuilderConfiguration({
+      setupQueryBuilderState: () => {
+        const queryBuilderState = new DataSpaceQueryBuilderState(
```

Comment:
lets add the config, disableCompile, we should do this for most things. i want to change even doing it directly from service editor to not require compilation. Thanks